### ISSUE TITLE
Achievements fail to unlock

### ISSUE THREAD

**Issue by programmermattsrum** [playgameservices/play-games-plugin-for-unity#2905]
I am having an issue with achievements unlocking for both incremental and non-incremental.  For non-incremental I am just reporting 100.0f progress and for incremental I am reporting the steps to the API.  Each increment is working fine accept the last step.  For example if there is 10 steps.  It is incrementing to 9 steps just fine but then on step 10 it is not getting changed from 9 to 10 and unlocking.  Each time I try to increment it says it is incrementing by 1 step but it doesn't show in the Play Games App as being completed still 90% done.  

I also tried using UnlockAchievement from the API and that wasn't working either.

I am using the following version below
Unity 2018.4.23.f1
Play Games Plugin 0.10.09

```
        private void Report(string id, PlayGamesAchievement a, int metric)
        {
            Debug.LogFormat("Reporting Achievement: {0}", a.title);
            
            if (a.hidden)
            {
                Social.ReportProgress(id, 0.0f, b => { });
                Debug.LogFormat("Achievement Revealed: {0}", a.title);
            }

            if (a.isIncremental)
            {
                if ((metric > a.currentSteps) && !a.completed)
                {
                    var delta = metric - a.currentSteps;
                    Debug.Log( "Metric " + a.title + ": " + metric);
                    Debug.LogFormat("Incremented Achievement: {0} Before({1}/{2})", a.title, a.currentSteps, a.totalSteps);
                    if ((a.currentSteps + delta) > a.totalSteps)
                    {
                        delta = a.totalSteps - a.currentSteps;
                    }
                    PlayGamesPlatform.Instance.IncrementAchievement(id, delta, b => {});
                    if ((a.currentSteps + delta) == a.totalSteps)
                    {
                        Social.ReportProgress(id, 100.0f, b => { });
                    }
                }
            }
            else
            {
                if (metric == 1 && !a.completed)
                {
                    Social.ReportProgress(id, 100.0f, b => { });
                    Debug.LogFormat("Achievement Completed: {0}", a.title);
                }
            }
        }
```

Logs from the API showing that it is taking the step and marking as 100.0f completed

2020-06-18 13:10:54.615 I/Unity:  [Play Games Plugin 0.10.09] 06/18/20 13:10:54 -05:00 DEBUG: IncrementAchievement: CgkI5KaW598XEAIQLA, steps 1
    GooglePlayGames.OurUtils.PlayGamesHelperObject:Update()
  
2020-06-18 13:10:54.616 I/Unity:  [Play Games Plugin 0.10.09] 06/18/20 13:10:54 -05:00 DEBUG: ReportProgress, CgkI5KaW598XEAIQLA, 100
    GooglePlayGames.OurUtils.PlayGamesHelperObject:Update()

**Comment by olehkuznetsov** [comment]
Seems it could be rounding error on our side here:
int targetSteps = (int) Math.Round((progress / 100f) * ach[i].TotalSteps);

can you try to pass 110.0f instead to unlock incremental achievement, please?

**Comment by programmermattsrum** [comment]
@olehkuznetsov I tried it with 100.0f and 110.0f but still didn't unlock the achievement.  It was my understanding that once the achievement reached the total steps it would unlock.  That doesn't seem to be happening.  

Master 2 levels = 2 steps total (Stuck at 1 step) - passing 1 step does not complete it.  Passing 110.0f to reportProgress did not complete it either.
RevealAchievement: CgkI5KaW598XEAIQLA
IncrementAchievement: CgkI5KaW598XEAIQLA, steps 1
ReportProgress, CgkI5KaW598XEAIQLA, 110

Master 10 levels = 10 steps total (Stuck at 9 steps) - passing 1 step does not complete it. Passing 110.0f to reportProgress did not complete it either.
RevealAchievement: CgkI5KaW598XEAIQLQ
IncrementAchievement: CgkI5KaW598XEAIQLQ, steps 1
ReportProgress, CgkI5KaW598XEAIQLQ, 110

25 stars = 25 steps total (Stuck at 24 steps) - passing 1 step does not complete it. Passing 110.0f to reportProgress did not complete it either. 
RevealAchievement: CgkI5KaW598XEAIQNQ
IncrementAchievement: CgkI5KaW598XEAIQNQ, steps 1
ReportProgress, CgkI5KaW598XEAIQNQ, 110

I am also seeing with non-incremental achievements that they don't complete either when reporting progress of 100.0f or 110.0f

Get a gold medal passed 100.0f and 110.0f and was not unlocked
RevealAchievement: CgkI5KaW598XEAIQMA

**Comment by programmermattsrum** [comment]
@olehkuznetsov I did some more testing over the weekend.  I got mixed results.  I was able to complete the achievements but then when I completed another achievement every completed achievement went back to not completed and basically in the state it was before the achievements got completed.  This kept happening back and forth.  Very confusing results.

**Comment by olehkuznetsov** [comment]
That's a different issue. For that one can you check if you linked Android app in play console, please? the same package name/SHA1 pair as in cloud console.

**Comment by programmermattsrum** [comment]
@olehkuznetsov The package is already linked to the play console and the firebase using the game services as well.

**Comment by olehkuznetsov** [comment]
Can you share your package name, please if it is already published?

**Comment by programmermattsrum** [comment]
com.draykunstudios.LabChaos
Just so we are clear the achievements are not published yet but the app has already been downloaded 176k+ times.  We are bringing achievements in the next update.

**Comment by programmermattsrum** [comment]
I could add you as an internal tester if that would help.

**Comment by olehkuznetsov** [comment]
test.oneup172@gmail.com
can you add this one, please?

**Comment by programmermattsrum** [comment]
You have been added.

https://play.google.com/apps/internaltest/4700131273090109717

**Comment by programmermattsrum** [comment]
@olehkuznetsov Were you able to get the game installed and start testing it?

**Comment by olehkuznetsov** [comment]
Yes, I don't see anything suspicious in logs. Achievements weren't reseted for me. Do you have any specific repro steps for reseted achievements?

**Comment by programmermattsrum** [comment]
It only resets after I have completed one.  Did you complete the 25 plays or master 2 levels or 1 gold medal?  After that play another level and they will all go back to being not completed.

Sent from Yahoo Mail on Android 
 
  On Tue, Jun 23, 2020 at 10:45 AM, olehkuznetsov<notifications@github.com> wrote:   


Yes, I don't see anything suspicious in logs. Achievements weren't reseted for me. Do you have any specific repro steps for reseted achievements?

**Comment by olehkuznetsov** [comment]
Have you removed achievements button from the game? Can't find it anymore in the settings.

**Comment by programmermattsrum** [comment]
Sorry I had to fix a policy issue with the game.  I will get the game updated tomorrow for you.

**Comment by programmermattsrum** [comment]
@olehkuznetsov You should see the achievements button now.

**Comment by olehkuznetsov** [comment]
Thank you. I've reproduced it.

**Comment by olehkuznetsov** [comment]
Seems we have an issue with unlocking achievements without image set. You can add an unlocked image to 'Master 2 Levels ' achievement and it should fix the issue. 
We will fix it in one of the next Play games releases.

**Comment by programmermattsrum** [comment]
@olehkuznetsov I will give that a try and let you know the results.

**Comment by programmermattsrum** [comment]
@olehkuznetsov I had one more question.  Does the pop-up saying you unlocked the achievement only work for published achievements?

**Comment by olehkuznetsov** [comment]
Yes, it works for unpublished achievement. 
In this case it's the same issue with missing achievement image.

**Comment by programmermattsrum** [comment]
@olehkuznetsov Thanks for the help!   We published all the achievements in game services and the game is now live with achievements now.

**Comment by ajeets1978** [comment]
Social Plugin doesn't work for Incrementing the achievements
